### DISCUSSION TITLE
How to get multiple XMP blocks?

### DISCUSSION THREAD

**Discussion by hackerfactor** [Exiv2/exiv2#3003] [Q&A] [answered]
I have a file with multiple independent XMP blocks. How can I use libexiv2 to literate through them?

**Comment by clanmills** [comment, marked as answer]
I don't believe that's possible.  Adobe provide the specification for embedding XMP in images.  Exiv2 uses an elderly version of Adobe's XMPsdk which doesn't support multiple XMP blocks, nor extended XMP.

**Reply by hackerfactor** [reply]
You are correct that these files were not created using Adobe's library.

However, other applications occasionally write separate XMP blocks. I have plenty of examples where it happens. I've seen:
1. Applications that write their own XMP block because they either don't know or don't care about the previous XMP block. (Easier to create than to try to append.)
2. Cameras that write an XMP block in the main JPEG, separate XMP block in nested thumbnail/preview images.
3. Extended XMP blocks (usually from Google's AI system).
4. Streaming videos where there's a periodic XMP block inserted with information (from a dashboard camera; it was writing GPS data as it streamed).

**Reply by clanmills** [reply]
@hackerfactor Well, Dr K, you can guess what I'm going to say.  If you think this is important, I invite you to join the team and work on this topic.  You'll find the people are friendly, supportive and clever.  I believe somebody is actively working to bring new versions of Adobe's XMPsdk into the code base, so you'll have company and get help.

I had a discussion in August with somebody using a Pixel Pro 7 phone with multiple XMP blocks.  https://github.com/Exiv2/exiv2/issues/3016

That issue was closed as `enhancement`.  There's always new and interesting stuff on which to work!  However, after 15 years of effort (and 15,000 hours), I've retired and happy to the project's _**grey beard.**_

**Reply by hackerfactor** [reply]
Sorry to hear that, but I understand. Enjoy the retirement!
